### PR TITLE
Document moduleRoutes hook

### PR DESCRIPTION
**Pull request by thomasnares**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       |  8.x
| Description?  | Add an example of hook moduleRoutes + details on implementation

### CODE REVIEW

File: modules/concepts/hooks/list-of-hooks/moduleRoutes.md [hunk 1]
```diff
@@ -29,4 +37,104 @@ This hook has a `$check_exception` parameter set to `false` (check permission ex
...
+  'params' => [
+    'fc' => 'module',
+    'module' => 'mymodulename'
+  ]
+]
+```
+
+### Implementation
+
+This example adds two routes for your module, one for listing items, the other one for showing one specific item:
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks/moduleRoutes.md]: this one is not accurate enough, you've shown how to declare the array, but maybe you should also mention ModuleFrontControllers that developers have to create. It'd be great to provide a full example for this particular hook

File: modules/concepts/hooks/list-of-hooks/moduleRoutes.md [hunk 2]
```diff
@@ -9,12 +9,20 @@ locations:
...
 ---
 
 # Hook moduleRoutes
 
 ## Information
 
+{{% notice tip %}}
+**Adds route to the PrestaShop router** 
+
+This hook allows your module to add routes to the PrestaShop router with parameters, wildcards, ...
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks/moduleRoutes.md]:
```suggestion
This hook allows your module to extend default PrestaShop routes with custom ones and map them to your module front controllers.
```

File: modules/concepts/hooks/list-of-hooks/moduleRoutes.md (hunk 1 above)
Comment (thomasnares) [reply modules/concepts/hooks/list-of-hooks/moduleRoutes.md]: ok, will improve it 👍

File: modules/concepts/hooks/list-of-hooks/moduleRoutes.md [hunk 3]
```diff
@@ -29,4 +37,154 @@ This hook has a `$check_exception` parameter set to `false` (check permission ex
...
+                'fc' => 'module',
+                'module' => 'mymoduleaddingroutes'
+            ]
+          ]
+        ];
+    }
+}
+```
+
+The complete implementation example is available in our [example modules repository](https://github.com/PrestaShop/example-modules).
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks/moduleRoutes.md]:
```suggestion
The complete implementation example is available in our [example modules repository](https://github.com/PrestaShop/example-modules/demomoduleroutes).
```

File: modules/concepts/hooks/list-of-hooks/moduleRoutes.md [hunk 4]
```diff
@@ -29,4 +37,154 @@ This hook has a `$check_exception` parameter set to `false` (check permission ex
...
+```php
+class MyModuleAddingRoutesShowModuleFrontController extends ModuleFrontController
+{
+    public function initContent()
+    {
+        $this->context->smarty->assign(
+            array(
+              'id' => Tools::getValue('id'),
+              'slug' => Tools::getValue('slug')
+            ));
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks/moduleRoutes.md]:
```suggestion
        // It is just an example. Remember to always validate the input data!
        $this->context->smarty->assign(
            [
              'id' => Tools::getValue('id'),
              'slug' => Tools::getValue('slug')
            ]
        );
```

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
🎉

**Pull request merged**